drivers/gpio/gpio_nrfx.c

```diff
@@ -590,26 +584,20 @@ static int gpio_nrfx_init(const struct device *port)
 	static const struct gpio_nrfx_cfg gpio_nrfx_p##id##_cfg = {	\
 		.common = {						\
 			.port_pin_mask =				\
-			GPIO_PORT_PIN_MASK_FROM_DT_NODE(GPIO(id)),	\
+			GPIO_PORT_PIN_MASK_FROM_DT_INST(id),		\
 		},							\
-		.port = NRF_P##id,					\
-		.port_num = id						\
+		.port = (NRF_GPIO_Type *)DT_INST_REG_ADDR(id),		\
+		.port_num = DT_INST_PROP(id, port)			\
 	};								\
 									\
 	static struct gpio_nrfx_data gpio_nrfx_p##id##_data;		\
 									\
-	DEVICE_DT_DEFINE(GPIO(id), gpio_nrfx_init,			\
+	DEVICE_DT_INST_DEFINE(id, gpio_nrfx_init,			\
 			 NULL,						\
 			 &gpio_nrfx_p##id##_data,			\
 			 &gpio_nrfx_p##id##_cfg,			\
 			 POST_KERNEL,					\
-			 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,		\
-			 &gpio_nrfx_drv_api_funcs)
+			 CONFIG_GPIO_NRF_INIT_PRIORITY,			\
```

Comment:
Ideally there should be a driver category init level, e.g. `CONFIG_GPIO_INIT_PRIORITY`

Reply:
yes, there are priorities in other vendor specific kconfigs but not in this PR.

Reply:
This is what I've done for pinctrl: https://github.com/zephyrproject-rtos/zephyr/pull/37572/commits/bd05806ca1f70e83b274be4a9a487e9d40152db2 (I needed control of GPIO priority)

Reply:
ok, so it will be adapted in another PR, right?